### PR TITLE
Median calc with timestamp and multi-thread rmat generator

### PR DESCRIPTION
**Pull request by KIwabuchi**
Change the median calculation program to support time stamps

Support multi-thread in the rmat edge generator in bfs

### CODE REVIEW

**Comment by mcfadden8** [comment]
@mayagokhale, I am not sure who would like to be reviewing the changes that go into this repository.  Can you think of other possible folks I should add to the team as collaborators?

File: src/bfs/README.md [hunk 1]
```diff
@@ -29,7 +29,8 @@ see [Graph500, 3.2 Detailed Text Description](https://graph500.org/?page_id=12#s
 ./generate_edge_list -o /mnt/ssd/edge_list -v 20 -e $((2**20*16))
 ````
 This command generates a edge list file (/mnt/ssd/edge_list) which contains the edges of a SCALE 20 graph.
-In Graph 500, the number of edges of a graph is #vertives x 16 (16 is called 'edge factor').
+In Graph 500, the number of edges of a graph is #vertives x 16 (16 is called 'edge factor') as an undirected graph.
+Note that #edges generated by this generator is #vertives x 16 x 2 if -u option (generates edges for the both direction) is true, which is default.
```
Comment (mayagokhale) [inline src/bfs/README.md]: correct spelling of vertices.

File: src/bfs/rmat_edge_generator/generate_edge_list.cpp [hunk 2]
```diff
@@ -98,21 +141,39 @@ int main(int argc, char **argv) {
...
+    for (auto edge : rmat) {
+      edge_list_file << edge.first << " " << edge.second << "\n";
+      ++num_generated_edges;
+    }
+    edge_list_file.close();
   }
 
-  for (auto edge : rmat) {
-    edge_list_file << edge.first << " " << edge.second << "\n";
+  // Sanitary check
```
Comment (mayagokhale) [inline src/bfs/rmat_edge_generator/generate_edge_list.cpp]: change to Sanity check

File: src/bfs/rmat_edge_generator/generate_edge_list.cpp [hunk 3]
```diff
@@ -98,21 +141,39 @@ int main(int argc, char **argv) {
...
+  const size_t num_edges_supposed_to_generate = (rmat_option.generate_both_directions) ? rmat_option.edge_count * 2ULL : rmat_option.edge_count;
+  if (num_edges_supposed_to_generate != num_generated_edges) {
+    std::cerr << "The number of generated edges is wrong " << num_generated_edges << std::endl;
```
Comment (mayagokhale) [inline src/bfs/rmat_edge_generator/generate_edge_list.cpp]: add in the error message the number of edges that were supposed to have been generated so the numbers can be compared for debugging.

**Review by mayagokhale — APPROVED** [review]
Just a couple of spelling and minor comments to be updated. I will approve merging.

**Comment by mcfadden8** [comment]
@KeitaRckPn and @mayagokhale, since this PR was marked approved, I have merged it in.

**Pull request merged**